src/rpc/RPCHelpers.cpp

```diff
@@ -904,14 +904,18 @@ xrpLiquid(
     ripple::SLE const sle{it, key};
 
     std::uint32_t const ownerCount = sle.getFieldU32(ripple::sfOwnerCount);
-
     auto const reserve = backend.fetchFees(sequence, yield)->accountReserve(ownerCount);
```

Comment:
Since reserve is used conditionally, we can save one backend call if it is looking for amm

Reply:
Hey @cindyyan317 , that is a good point.
There's 2 options: 
1. putting the reserve inside the lambda like this
```
ripple::STAmount const amount = [&]() {
        // AMM doesn't require the reserves
        if ((sle.getFlags() & ripple::lsfAMMNode) != 0u)
            return balance;
        auto const reserve = backend.fetchFees(sequence, yield)->accountReserve(ownerCount);
        ripple::STAmount amount = balance - reserve;
        if (balance < reserve)
            amount.clear();
        return amount;
    }();
```
Pros: No extra check required and saves one call if it is looking for amm
Cons: It is inside the lambda, in this case lambda is just called once so it's fine but if it is called multiple times - we will be doing multiple calls.

2. Just adding a simple if statement check to do a backend fetch for reserve
```
if ((sle.getFlags() & ripple::lsfAMMNode) == 0u)
  reserve = backend.fetchFees(sequence, yield)->accountReserve(ownerCount);
```
Pros: Outside lambda and saves one backend call in case of amms
Cons: Now there's two checks instead one.

What do you think ? I'm leaning towards the second option

Reply:
I think both are okay.  maybe simple if is more readable.

Reply:
Sounds good, thank you. Marking this as resolved!